flink-ml-lib/src/test/java/org/apache/flink/ml/clustering/AgglomerativeClusteringTest.java

```diff
@@ -203,6 +229,59 @@ public void testTransform() throws Exception {
                 agglomerativeClustering.getPredictionCol());
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTransformWithEventTimeWindow() throws Exception {
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        new TypeInformation<?>[] {DenseVectorTypeInfo.INSTANCE, Types.INSTANT},
+                        new String[] {"features", "ts"});
+
+        Instant baseTime = Instant.now();
+        DataStream<Row> inputDataStream =
+                env.fromCollection(INPUT_DATA)
+                        .setParallelism(1)
+                        .map(x -> Row.of(x, baseTime.plusSeconds((long) x.get(0))), outputTypeInfo);
+
+        Schema schema =
+                Schema.newBuilder()
+                        .column("features", DataTypes.of(DenseVectorTypeInfo.INSTANCE))
+                        .column("ts", DataTypes.TIMESTAMP_LTZ(3))
+                        .watermark("ts", "ts - INTERVAL '5' SECOND")
+                        .build();
+
+        Table inputDataTable = tEnv.fromDataStream(inputDataStream, schema);
+
+        AgglomerativeClustering agglomerativeClustering =
+                new AgglomerativeClustering()
+                        .setLinkage(AgglomerativeClusteringParams.LINKAGE_AVERAGE)
+                        .setDistanceMeasure(EuclideanDistanceMeasure.NAME)
+                        .setPredictionCol("pred")
+                        .setWindows(EventTimeTumblingWindows.of(Time.seconds(1)));
```

Comment:
Can we keep the python test and java test consistent? Maybe add a test with CountTumblingWindow in Java and add a test with EventTimeTumblingWindows in Python?